pig/src/main/java/com/netflix/iceberg/pig/IcebergInputFormat.java

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.iceberg.pig;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.netflix.iceberg.CombinedScanTask;
+import com.netflix.iceberg.DataFile;
+import com.netflix.iceberg.FileScanTask;
+import com.netflix.iceberg.PartitionSpec;
+import com.netflix.iceberg.Schema;
+import com.netflix.iceberg.Table;
+import com.netflix.iceberg.TableScan;
+import com.netflix.iceberg.expressions.Expression;
+import com.netflix.iceberg.hadoop.HadoopInputFile;
+import com.netflix.iceberg.io.InputFile;
+import com.netflix.iceberg.parquet.Parquet;
+import com.netflix.iceberg.types.TypeUtil;
+import com.netflix.iceberg.types.Types;
+import org.apache.commons.lang.SerializationUtils;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.parquet.hadoop.api.ReadSupport;
+import org.apache.pig.impl.util.ObjectSerializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static com.netflix.iceberg.pig.SchemaUtil.*;
+
+public class IcebergInputFormat<T> extends InputFormat<Void, T> {
```

Comment:
Should this be named IcebergPigInputFormat since it only works for Pig?

Reply:
It's in a 'pig' package, so it could be argued that is redundant.  Ideally, we would find a way to abstract the few ties to pig and promote it.  If you feel strongly about it, I'll make the change, but seems unnecessary.

Reply:
I think it would help because it will be shown in Javadoc without the package name, and an IDE can import IcebergInputFormat and fill in the package automatically. By changing the class name, I think we would have less confusion.